File: app/src/main/java/io/github/hyuwah/catatanku/editor/EditorMarkdownActivity.java
package io.github.hyuwah.catatanku.editor;

import android.os.Bundle;
import android.text.TextUtils;

import androidx.appcompat.app.AppCompatActivity;

import br.tiagohm.markdownview.MarkdownView;
import br.tiagohm.markdownview.css.styles.Github;
import io.github.hyuwah.catatanku.R;

public class EditorMarkdownActivity extends AppCompatActivity {

  MarkdownView markdownView;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_editor_markdown);

    markdownView = findViewById(R.id.editor_markdown_view);

    Bundle bundle = getIntent().getExtras();
    String mTitle = bundle.getString("TITLE");
    String mBody = bundle.getString("BODY");

    // TODO make bundle into variable, handle lifecycle change

    // TODO Handle Empty view?
    if(TextUtils.isEmpty(mTitle)){
      setTitle("Untitled");
    }else{
      setTitle(mTitle);
    }

    markdownView.addStyleSheet(new Github());
    markdownView.loadMarkdown(bundle.getString("BODY"));

  }
}

File: app/src/main/java/io/github/hyuwah/catatanku/ui/editor/EditorMarkdownActivity.kt
package io.github.hyuwah.catatanku.ui.editor

import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.foundation.gestures.Orientation
import androidx.compose.foundation.gestures.scrollable
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Scaffold
import androidx.compose.runtime.getValue
import androidx.compose.runtime.livedata.observeAsState
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import dagger.hilt.android.AndroidEntryPoint
import dev.jeziellago.compose.markdowntext.MarkdownText
import io.github.hyuwah.catatanku.ui.common.DefaultAppToolBar
import io.github.hyuwah.catatanku.ui.theme.AppTheme

@AndroidEntryPoint
class EditorMarkdownActivity : AppCompatActivity() {
    private val viewModel: EditorViewModel by viewModels()

    private val noteId by lazy {
        intent.getStringExtra("ID").orEmpty()
    }
    private val legacyTitle by lazy {
        intent.getStringExtra("TITLE").orEmpty()
    }
    private val legacyContent by lazy {
        intent.getStringExtra("BODY").orEmpty()
    }

    @OptIn(ExperimentalMaterial3Api::class)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (noteId.isNotBlank()) {
            viewModel.getNoteById(noteId)
        }
        setContent {
            val note by viewModel.note.observeAsState()
            AppTheme {
                Scaffold(
                    topBar = {
                        DefaultAppToolBar(
                            text = legacyTitle.ifBlank { note?.title.orEmpty() }.ifBlank { "Untitled" },
                            onNavClicked = { finish() }
                        )
                    }
                ) { paddingValues ->
                    Column(
                        modifier = Modifier
                            .padding(paddingValues)
                            .fillMaxSize()
                            .scrollable(rememberScrollState(), orientation = Orientation.Vertical),
                    ) {
                        MarkdownText(
                            markdown = legacyContent.ifBlank { note?.contentText.orEmpty() },
                            modifier = Modifier
                                .fillMaxSize()
                                .padding(12.dp)
                        )
                    }
                }
            }
        }
    }
}